api/api.go

```diff
@@ -19,32 +19,51 @@ import (
 	yaml "gopkg.in/yaml.v2"
 )
 
-// GQLResponseErrors is a slice of errors returned by the GraphQL server. Each
-// error message is a key-value pair with the structure "Message: string"
-type GQLResponseErrors struct {
-	Errors []struct {
-		Message string
+// GQLErrorsCollection is a slice of errors returned by the GraphQL server.
+// Each error is made up of a GQLResponseError type.
+type GQLErrorsCollection []GQLResponseError
+
+// Error turns a GQLErrorsCollection into an acceptable error string that can be printed to the user.
+func (errs GQLErrorsCollection) Error() string {
+	messages := []string{}
+
+	for i := range errs {
+		messages = append(messages, errs[i].Message)
 	}
+
+	return strings.Join(messages, ": ")
```

Comment:
Joining errors with a colon is a convention used to wrap an error with additional context as it passes through callers.

In this case it's being used to concat multiple errors, which seems very different. This could result in some very confusing errors. It might be better to join these errors with newlines or commas.